homework/homework24/tests/test_web_page.py

```diff
@@ -0,0 +1,196 @@
+
+    submit_button = wait.until(EC.element_to_be_clickable(
+        (By.CSS_SELECTOR, AddContactLocators.submit_btn)))
+    submit_button.click()
+
+    contact_row = wait.until(EC.presence_of_element_located(
+        (By.XPATH, EditContactLocators.contact_row)))
+
+    # Verify that the contact was added
+    assert contact_row is not None
+
+
+def test_edit_contact(open_browser):
+    """
+    Edits details of a contact on a web page.
+
+    Parameters:
+        open_browser (selenium.webdriver.Chrome):
+        The browser fixture to open the login page.
+    """
+    browser = open_browser
+    wait = WebDriverWait(browser, 10)
+
+    contact_row = wait.until(EC.element_to_be_clickable(
+        (By.XPATH, EditContactLocators.contact_row)))
+    contact_row.click()
+
+    button_edit = wait.until(EC.element_to_be_clickable(
+        (By.CSS_SELECTOR, EditContactLocators.edit_contact_btn)))
+    button_edit.click()
+    time.sleep(1)
+
+    # Prepare updated data for a contact
+    updated_data = {
+        EditContactLocators.email_form: 'newtestmail@beispiel.de',
+        EditContactLocators.city_form: 'Dortmund',
+        EditContactLocators.postal_code_form: '44123',
+    }
+
+    # Fill in the form fields and wait until each field is available
+    for locator, value in updated_data.items():
+        input_field = wait.until(EC.element_to_be_clickable(
+            (By.CSS_SELECTOR, locator)))
+        input_field.clear()
+        input_field.send_keys(value)
+
+    submit_button = wait.until(EC.element_to_be_clickable(
+        (By.CSS_SELECTOR, EditContactLocators.submit_btn)))
+    submit_button.click()
+    time.sleep(1)
+
+    # Verify that the contact was edited
+    email = browser.find_element(
+        By.CSS_SELECTOR, '[id="email"]').text
+    city = browser.find_element(
+        By.CSS_SELECTOR, '[id="city"]').text
+    postal_code = browser.find_element(
+        By.CSS_SELECTOR, '[id="postalCode"]').text
+    assert email == 'newtestmail@beispiel.de'
+    assert city == 'Dortmund'
+    assert postal_code == '44123'
+
+
+def test_delete_contact(open_browser):
+    """
+    Deletes a contact on a web page.
+
+    Parameters:
+        open_browser (selenium.webdriver.Chrome):
+        The browser fixture to open the login page.
+    """
+    browser = open_browser
+    wait = WebDriverWait(browser, 10)
+
+    contact_row = wait.until(EC.presence_of_element_located(
+        (By.XPATH, EditContactLocators.contact_row)))
+    # Save the name of the contact before deleting it
+    contact_name = contact_row.text
+    contact_row.click()
+
+    button_delete = wait.until(EC.element_to_be_clickable(
+        (By.CSS_SELECTOR, EditContactLocators.delete_contact_btn)))
+    button_delete.click()
+
+    alert = wait.until(EC.alert_is_present())
+    alert.accept()
+
+    contact_deleted = True
+    try:
+        wait.until(EC.presence_of_element_located(
+            (By.XPATH, f"//*[contains(text(), '{contact_name}')]")))
```

Comment:
Can be in locators as well:
my_locator = "//*[contains(text(), '{contact_name}')]"
```suggestion
            (By.XPATH, my_locator.format(my_locator))))
```

Reply:
fixed, add variables to file with locators and change line